com.playeveryware.eos/Runtime/Core/Common/SetOfNamed.cs

```diff
@@ -184,6 +205,24 @@ public bool Contains(T item)
 
             return false;
         }
+
+        public new bool Remove(Named<T> item)
+        {
+            // If a predicate is not defined, or the predicate returns false,
+            // then return false and stop.
+            if (_removePredicate != null && !_removePredicate(item.Value))
+            {
+                return false;
+            }
+            
+            // Remove the event subscription
+            item.NameChanged -= OnItemNameChanged;
+
+            // Otherwise attempt to remove the item
+            base.Remove(item);
```

Comment:
Should this return the result of this function? `base.Remove` makes me think that it has the same return type as this function, which is `bool`. So should that pass through in the result?

Reply:
Discussed this in meeting - however, adding some more comments to describe how and why this works would be good.